docs/bootstrap.json

```diff
@@ -0,0 +1,20 @@
+{
+  "ID": "bootstrap",
+  "Properties": {
+    "Resources": {
+      "resource_a": {
+        "Plugin": "resource-file",
+        "Type": "a",
+        "Properties": {
+          "Note": "Depends on {{.Resources.resource_b.ID}}"
```

Comment:
FWIW, being able to interpolate strings directly (vs. `Ref` style) can be pretty nice, since you might want to do something like 

```json
{
    "Name": "{{.SwarmName}}-VPC-{{.Count}}"
}
``` 

to concat strings, whereas Azure ends up looking something like 

```json
{
    "Name": "[concat(variables('SwarmName'), 'VPC', count())]"
}
```

which I find a lot less readable.  The tricky part is to determine how nested you'll get, since [Terraform for instance](https://www.terraform.io/docs/configuration/interpolation.html) has somewhat a problem where these giant monstrosities of `join(split('/', concat(blah, blah, blah)))` get built up, and a nested way to do it might be nicer...  In that case, they partially have that issue because you need to use the string interpolation syntax to return other types such as lists.  I find their use of some things like `*` really hard to reason about as well...

Ultimately I think it all depends on use case.  Might be kinda nice to work towards getting a "short-hand" syntax for exclusively strings but also have `Ref` available when you need to do something slightly more complicated.